aws_lambda_powertools/utilities/parser/functions.py

```diff
@@ -82,3 +82,28 @@ def _parse_and_validate_event(data: dict[str, Any] | Any, adapter: TypeAdapter):
             data = json.loads(data)
 
     return adapter.validate_python(data)
+
+
+def _validate_source_ip(value):
+    """
+    Handle sourceIp that may come with port (e.g., "10.1.15.242:39870")
+    in certain network configurations like Cloudflare + CloudFront + API Gateway.
+    Validates the IP part while preserving the original format.
+    See: https://github.com/aws-powertools/powertools-lambda-python/issues/7288
+    """
+
+    if value == "test-invoke-source-ip":
+        return value
+
+    try:
```

Comment:
In the discussion it was mentioned that `value` could be a CIDR, does `IPvAnyNetwork` cover that?

Reply:
Yes, it covers any IP address with CIDR. If you notice, this is the original type supported in the model.